### ISSUE TITLE
Ability to filter out or regex include form elements that are used in spider node generation.

### ISSUE THREAD

**Issue by JamieAzure** [zaproxy/zaproxy#5093]
The site I'm working on has dynamically named CSRF Tokens.  For example:

<input type="hidden" name="token_8570C740-BCFC-8C59-B7F2377E9885D42D" value = "8570C741-C95E-F5BB-71AA3EA5FD8F4E7F">

I can see where to add the CSRF here:

![image](https://user-images.githubusercontent.com/841771/47714500-57442680-dc3d-11e8-99c9-d1651ae0caed.png)

but it doesn't seem to take the wild carding regex aspect of the FORM field name.  Is there anyway to handle this in the current tools?

If that's not an option then I'd need a way to tell the spider that all of these are not unique nodes:
![image](https://user-images.githubusercontent.com/841771/47714451-38459480-dc3d-11e8-9439-be3d242a513d.png)

I'm unable currently to spider the site as these form elements are on nearly every page and the spider just loses its mind thinking there are millions of variations of a single page.

**Comment by setzamora** [comment]
@thc202 is it okay for me to work on this?

**Comment by thc202** [comment]
Yes, but first we need to discuss how we want this to work taking into account #4388 (which is already being worked on, in #4398), there's a great overlap between the two issues (e.g. both "filters" can be managed through the same UI).

**Comment by setzamora** [comment]
Initially, I'm thinking of adding regex support for each section under _Options_ (_Anti-CSRF Tokens, HTTP Sessions_, and _Spider_)

For example
![image](https://user-images.githubusercontent.com/18208623/48298425-45059a80-e511-11e8-9219-fb1b0c325478.png)

The token names will be similar to how regex is supported on the _Active Scan Input Vectors_ options
![image](https://user-images.githubusercontent.com/18208623/48298446-c3623c80-e511-11e8-9138-0468bad33987.png)

We'll convert these user interfaces (screenshots taken from #4398)
![image](https://user-images.githubusercontent.com/18208623/48298456-09b79b80-e512-11e8-80b6-4dc28b2b454b.png)

![image](https://user-images.githubusercontent.com/18208623/48298479-500cfa80-e512-11e8-9131-a2d1c2386470.png)

To something like
![image](https://user-images.githubusercontent.com/18208623/48298545-4fc12f00-e513-11e8-9e1a-14a4faa19da5.png)

For this phase, I reckon, it is practical to retain each in their own separate sections i.e. in the Options dialog. I don't find it confusing having it that way from a UX standpoint personally. Regardless, I'm open to whichever direction we'll take. 

One thing I haven't checked yet is the impact of these to the ZAP API and etc. Let me know your thoughts.

Cheers

**Comment by JamieAzure** [comment]
Was checking in on progress for this feature.  Any progress, or should I look elsewhere at this point for a scanning tool until this is completed?

**Comment by kingthorin** [comment]
> Unfortunately ZAP is an Open Source project supported and developed by volunteers and the community. There is currently no one assigned to the issue you raised, and I'm not aware of anyone discussing tackling it in the near or mid-term.

> There are two things you can try:
> 1) Become a contributor, we're always glad to have code from new people.
> 2) You could support the specific issue via Bounty Source and hope that someone might pick it up and tackle it on your behalf: https://www.bountysource.com/issues/65382027-ability-to-filter-out-or-regex-include-form-elements-that-are-used-in-spider-node-generation

**Comment by kingthorin** [comment]
@setzamora that seems reasonable to me (sorry thus slipped off my radar). @psiinon @thc202 What do you think?

**Comment by psiinon** [comment]
I'm good with that.

**Comment by thc202** [comment]
@setzamora with ` same UI` I was suggesting using the same UI class not use just one options panel for all, sorry for not being clear. Otherwise looks good.

**Comment by thc202** [comment]
This should be done in the add-on now, located in the `zap-extensions` repo.
